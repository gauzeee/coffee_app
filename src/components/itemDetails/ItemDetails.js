import React, { Component } from "react";
import Separator from "../separator";
import Spinner from "../spinner";
import ErrorMessage from "../errorMessage";

class ItemDetails extends Component {
  state = {
    item: null,
    loading: true,
    error: false
  };

  componentDidMount() {
    const { item } = this.props.location.state;
    this.setState({
      item,
      loading: false
    });
  }

  componentDidCatch() {
    this.setState({
      error: true
    });
  }

  render() {
    if (this.state.error) return <ErrorMessage />;
    const { loading } = this.state;
    if (loading) return <Spinner />;
    let { url, name, country, desc, price } = this.state.item;

    country = country ? country : "We don`t know about it...";
    desc = desc ? desc : "Coming soon!";

    return (
      <div className="item-info">
        <div className="item-image">
          <img src={url} alt="" />
        </div>
        <div className="item-details">
          <h3>{name}</h3>
          <Separator light="black" />
          <p className="item-country">
            <span className="type">Country:</span>
            <span className="value">{country}</span>
          </p>
          <p className="item-desc">
            <span className="type">Description:</span>
            <span className="value">{desc}</span>
          </p>
          <p className="item-price">
            <span className="type">Price:</span>
            <span className="value">{price}</span>
          </p>
        </div>
      </div>
    );
  }
}

export default ItemDetails;
