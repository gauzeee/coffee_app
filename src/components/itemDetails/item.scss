.item {
    &-info {
        display: flex;
        max-width: 800px;
        margin-left: auto;
        margin-right: auto;
        min-height: 455px;
        align-items: center;
    }

    &-image {
        width: 50%;
        text-align: right;
        margin-right: 42px;

        img {
            width: 70%;
        }
    }

    &-details {
        width: 50%;

        .item-country {
            margin-top: 40px;
        }

        p {
            font-size: 14px;
            text-align: left;
            color: #000000;
            margin-top: 16px;
            margin-bottom: 16px;

            .type {
                font-weight: bold;
                margin-right: 8px;
            }
        }
    }

    &-price {
        .type {
            font-size: 16px;
        }

        .value {
            font-size: 20px;
        }
    }
}