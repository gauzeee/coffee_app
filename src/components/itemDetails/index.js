import ItemDetails from "./ItemDetails";
import "./item.scss";

export default ItemDetails;
