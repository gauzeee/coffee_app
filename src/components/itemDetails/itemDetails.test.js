import React from "react";
import itemDetails from "./ItemDetails";
import { shallow } from "enzyme";
import ItemDetails from "./ItemDetails";

describe("Testing itemDetails", () => {
  const store = { state: { item: { name: "Test" } } };
  const item = shallow(<ItemDetails location={store} />);
  describe("Snap and state", () => {
    it("ItemDetails has rendered correctly", () => {
      expect(item).toMatchSnapshot();
    });
    it("ItemDetails item in state is null", () => {
      expect(item.state().item.name).toEqual("Test");
    });
    it("ItemDetails loading in state is true", () => {
      expect(item.state().loading).toBeFalsy();
    });
    it("ItemDetails error in state is false", () => {
      expect(item.state().error).toBeFalsy();
    });
  });
  describe("itemDetails catch testing", () => {
    const newItem = shallow(<ItemDetails location={store} />);
    newItem.instance().componentDidCatch();
    expect(newItem.state().error).toBeTruthy();
  });
});
