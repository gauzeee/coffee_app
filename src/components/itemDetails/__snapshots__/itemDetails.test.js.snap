// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing itemDetails Snap and state ItemDetails has rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemDetails
    location={
      Object {
        "state": Object {
          "item": Object {
            "name": "Test",
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="item-image"
        >
          <img
            alt=""
          />
        </div>,
        <div
          className="item-details"
        >
          <h3>
            Test
          </h3>
          <Separator
            light="black"
          />
          <p
            className="item-country"
          >
            <span
              className="type"
            >
              Country:
            </span>
            <span
              className="value"
            >
              We don\`t know about it...
            </span>
          </p>
          <p
            className="item-desc"
          >
            <span
              className="type"
            >
              Description:
            </span>
            <span
              className="value"
            >
              Coming soon!
            </span>
          </p>
          <p
            className="item-price"
          >
            <span
              className="type"
            >
              Price:
            </span>
            <span
              className="value"
            />
          </p>
        </div>,
      ],
      "className": "item-info",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt=""
          />,
          "className": "item-image",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Test
            </h3>,
            <Separator
              light="black"
            />,
            <p
              className="item-country"
            >
              <span
                className="type"
              >
                Country:
              </span>
              <span
                className="value"
              >
                We don\`t know about it...
              </span>
            </p>,
            <p
              className="item-desc"
            >
              <span
                className="type"
              >
                Description:
              </span>
              <span
                className="value"
              >
                Coming soon!
              </span>
            </p>,
            <p
              className="item-price"
            >
              <span
                className="type"
              >
                Price:
              </span>
              <span
                className="value"
              />
            </p>,
          ],
          "className": "item-details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Test",
            },
            "ref": null,
            "rendered": "Test",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "light": "black",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="type"
                >
                  Country:
                </span>,
                <span
                  className="value"
                >
                  We don\`t know about it...
                </span>,
              ],
              "className": "item-country",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Country:",
                  "className": "type",
                },
                "ref": null,
                "rendered": "Country:",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "We don\`t know about it...",
                  "className": "value",
                },
                "ref": null,
                "rendered": "We don\`t know about it...",
                "type": "span",
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="type"
                >
                  Description:
                </span>,
                <span
                  className="value"
                >
                  Coming soon!
                </span>,
              ],
              "className": "item-desc",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Description:",
                  "className": "type",
                },
                "ref": null,
                "rendered": "Description:",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Coming soon!",
                  "className": "value",
                },
                "ref": null,
                "rendered": "Coming soon!",
                "type": "span",
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="type"
                >
                  Price:
                </span>,
                <span
                  className="value"
                />,
              ],
              "className": "item-price",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Price:",
                  "className": "type",
                },
                "ref": null,
                "rendered": "Price:",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "value",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="item-image"
          >
            <img
              alt=""
            />
          </div>,
          <div
            className="item-details"
          >
            <h3>
              Test
            </h3>
            <Separator
              light="black"
            />
            <p
              className="item-country"
            >
              <span
                className="type"
              >
                Country:
              </span>
              <span
                className="value"
              >
                We don\`t know about it...
              </span>
            </p>
            <p
              className="item-desc"
            >
              <span
                className="type"
              >
                Description:
              </span>
              <span
                className="value"
              >
                Coming soon!
              </span>
            </p>
            <p
              className="item-price"
            >
              <span
                className="type"
              >
                Price:
              </span>
              <span
                className="value"
              />
            </p>
          </div>,
        ],
        "className": "item-info",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt=""
            />,
            "className": "item-image",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Test
              </h3>,
              <Separator
                light="black"
              />,
              <p
                className="item-country"
              >
                <span
                  className="type"
                >
                  Country:
                </span>
                <span
                  className="value"
                >
                  We don\`t know about it...
                </span>
              </p>,
              <p
                className="item-desc"
              >
                <span
                  className="type"
                >
                  Description:
                </span>
                <span
                  className="value"
                >
                  Coming soon!
                </span>
              </p>,
              <p
                className="item-price"
              >
                <span
                  className="type"
                >
                  Price:
                </span>
                <span
                  className="value"
                />
              </p>,
            ],
            "className": "item-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Test",
              },
              "ref": null,
              "rendered": "Test",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "light": "black",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="type"
                  >
                    Country:
                  </span>,
                  <span
                    className="value"
                  >
                    We don\`t know about it...
                  </span>,
                ],
                "className": "item-country",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Country:",
                    "className": "type",
                  },
                  "ref": null,
                  "rendered": "Country:",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "We don\`t know about it...",
                    "className": "value",
                  },
                  "ref": null,
                  "rendered": "We don\`t know about it...",
                  "type": "span",
                },
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="type"
                  >
                    Description:
                  </span>,
                  <span
                    className="value"
                  >
                    Coming soon!
                  </span>,
                ],
                "className": "item-desc",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Description:",
                    "className": "type",
                  },
                  "ref": null,
                  "rendered": "Description:",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Coming soon!",
                    "className": "value",
                  },
                  "ref": null,
                  "rendered": "Coming soon!",
                  "type": "span",
                },
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="type"
                  >
                    Price:
                  </span>,
                  <span
                    className="value"
                  />,
                ],
                "className": "item-price",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Price:",
                    "className": "type",
                  },
                  "ref": null,
                  "rendered": "Price:",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "value",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
